File: ActiveProbe.Domain/Models/DefinedTest.cs
using System;
using System.Collections.Generic;

namespace ActiveProbe.Domain.Models
{
    public partial class DefinedTest
    {
        public DefinedTest()
        {
            DefinedTestMachine = new HashSet<DefinedTestMachine>();
            DefinedTestMachineGroup = new HashSet<DefinedTestMachineGroup>();
        }

        public int Id { get; set; }
        public string Title { get; set; }
        public bool IsActive { get; set; }
        public bool Layer3Messages { get; set; }
        public short RepeatTypeId { get; set; }
        public int? RepeatTime { get; set; }
        public int? RepeatCount { get; set; }
        public int MeasurementInterval { get; set; }
        public short TestTypeId { get; set; }
        public int? UsualCallDuration { get; set; }
        public int? UsualCallWaitTime { get; set; }
        public string UsualCallNumber { get; set; }
        public short? TestDataId { get; set; }
        public short? TestDataTypeId { get; set; }
        public string TestDataServer { get; set; }
        public string TestDataUserName { get; set; }
        public string TestDataPassword { get; set; }
        public string TestDataDownloadFileAddress { get; set; }
        public int? TestDataUploadFileSize { get; set; }
        public short? IptypeId { get; set; }
        public short? OttserviceId { get; set; }
        public short? OttserviceTestId { get; set; }
        public short NetworkId { get; set; }
        public short? BandId { get; set; }
        public bool? SaveLogFile { get; set; }
        public short? LogFilePartitionTypeId { get; set; }
        public short? LogFilePartitionTime { get; set; }
        public short? LogFilePartitionSize { get; set; }
        public int? LogFileHoldTime { get; set; }
        public short? NumberOfPings { get; set; }
        public short? PacketSize { get; set; }
        public short? InternalTime { get; set; }
        public int? ResponseWaitTime { get; set; }
        public short? Ttl { get; set; }
        public int? FileSize { get; set; }
        public int? TraceRouteHubCount { get; set; }

        public virtual TestType TestType { get; set; }
        public virtual ICollection<DefinedTestMachine> DefinedTestMachine { get; set; }
        public virtual ICollection<DefinedTestMachineGroup> DefinedTestMachineGroup { get; set; }
    }
}

File: ActiveProbLTE/ClientApp/src/app/Shared/models/definedTest.ts
export class DefinedTest {
    Id: number;
    IsActive: Boolean;
    Layer3Messages: Boolean;
    RepeatTypeId: number;
    RepeatTime: number;
    RepeatCount: number;
    MeasurementInterval: number;//always seconds
    TestTypeId: number;//UsualCall, MosCall, Idle, Data, OTTServices
    UsualCallDuration: number;//Second
    UsualCallWaitTime: number;//Second
    UsualCallNumber: number;//CellNumber
    TestDataId: number;//FTP, HTTP, Ping, TraceRoute,...
    TestDataTypeId: number;//Download, Upload 
    TestDataServer: string;//address of server
    TestDataUserName: string;//user name to login to server
    TestDataPassword: string;//password to login to server
    TestDataDownloadFileAddress: string;//name and address of file in server
    TestDataUploadFileSize: number;//limitation for size(kb) of file to upload to server 
    IPTypeId: number//IPV4 or IPV6
    OTTServiceId: number;//Skype, WhatsApp, Instagram
    OTTServiceTestId: number;//VoiceCall, SendText, Pic
    NetworkId: number;//GSM Only, WCDMA Only, LTE Only, GSM + WCDMA, ...
    BandId: number;//7600, ...
    SaveLogFile: Boolean;
    LogFilePartitionTypeId: number;//Time/Size
    LogFilePartitionTime: number;//min
    LogFilePartitionSize: number;//kb
    LogFileHoldTime: number;//Hour
    NumberOfPings: number;//1 to 100
    PacketSize: number//4 to 188
    InternalTime: number//1000 to 10000
    ResponceWaitTime: number//10000 to 100000
    TTL: number //16 to 255
    TraceRouteHubCount: number
    RepeatTypeTitle: string;
    TestTypeTitle: string;
    TestDataTypeTitle: string;
    OTTServiceTitle: string;
    OTTServiceTestTitle: string;
    NetworkTitle: string;
    BandTitle: string;
    LogFilePartitionTypeTitle: string;
    IPTypeTitle: string;
}
